README.md

```diff
@@ -19,8 +19,235 @@
+
+* `PERSISTENT_STATE_TARGET`: This is the folder where the persistent state data
+  will be stored, if any. Default value is `/usr/local/.state`.
+
+* `PERSISTENT_STATE_PATHS`: This is a space separated list of paths. These are
+  the paths that will become writable and store its data inside
+  `PERSISTENT_STATE_TARGET`. By default, this variable is empty, which means
+  no persistent state area is created or used.
+
+  **Note**: The specified paths needs either to exist or be located in an area
+  which is writeable ( for example, inside locations specified with `RW_PATHS`).
+  The dracut module will attempt to create non-existant directories,
+  but might fail if the mountpoint where are located is read-only.
+
+* `PERSISTENT_STATE_BIND="true|false"`: When this variable is set to true
+  the persistent state paths are bind mounted (instead of using overlayfs)
+  after being mirrored with the original content. By default, this variable is
+  set to `false`.
+
+Note that persistent state are is set up once the ephemeral paths and persistent
+volumes are mounted. Persistent state paths can't be an already existing mount
+point. If the persistent state requires any of the paths that are part of the
+ephemeral area by default, then `RW_PATHS` needs to be defined to avoid
+overlapping paths.
+
+For example a common cOS configuration can be expressed as part of the
+cloud-init configuration as follows:
+
+```yaml
+name: example
+stage:
+  rootfs:
+    - name: "Layout configuration"
+      environment_file: /run/cos/cos-layout.env
+      environment:
+        VOLUMES: "LABEL=COS_OEM:/oem LABEL=COS_PERSISTENT:/usr/local"
+        OVERLAY: "tmpfs:25%"
+```
+
+You can also see the default config that we provide in https://github.com/kairos-io/kairos/blob/master/overlay/files/system/oem/11_persistency.yaml
+
+## What is the default workflow of Immucore
+
+----
+
+It starts pretty early in the boot process, just after `systemd-udev-settle.service` and before `dracut-initqueue.service`
+To see the full bootup process from dracut you can check [here](https://man7.org/linux/man-pages/man7/dracut.bootup.7.html)
+
+Just after starting, Immucore mounts `/proc` if it's not mounted in order to read the `/proc/cmdline` and obtains the different stanzas in order to configure itself.
+After checking the cmdline, it knows in which path is being booted, either active/passive/recovery or Netboot/LiveCD/Do nothing.
+
+Based on that it builds a DAG with the steps needed to complete and process through the DAG until its completed. It also builds a `State` object which has all the configs needed to properly mount and configure the system properly.
+Once the DAG has been completed (and with no errors), Immucore its finished, and it's ready for the initramfs init process to do a switch_root and pivot into the final root to boot the system.
+
+When booting from Netboot/LiveCD/Do nothing (`rd.cos.disable` or `rd.immucore.disable` on the cmdline) the DAD It's pretty simple. 
```

Comment:
```suggestion
When booting from Netboot/LiveCD/Do nothing (`rd.cos.disable` or `rd.immucore.disable` on the cmdline) the DAG It's pretty simple. 
```

Reply:
I think my way its better. I mean, the Dads are pretty simple, I am at least :P